Gems/Atom/RHI/Metal/Code/Source/RHI/CommandList.cpp

```diff
@@ -447,57 +400,129 @@ namespace AZ
+        }
+    
+        void CommandList::CollectBindlessComputeUntrackedResources(const ShaderResourceGroup* shaderResourceGroup,
+                                                ArgumentBuffer::ResourcesForCompute& untrackedResourceComputeRead,
+                                                ArgumentBuffer::ResourcesForCompute& untrackedResourceComputeReadWrite)
+        {
+            if(shaderResourceGroup->GetData().GetBindlessViewsSize()==0)
             {
-                MakeHeapsResident(mtlRenderStagesForNullDescHeap);
+                return;
+            }
+            
+            const AZStd::vector<RHI::ShaderResourceGroupData::BindlessResourceViews> bindlessResourcesVec =
+                                    shaderResourceGroup->GetData().GetAllBindlessViews();
+            for(const auto& it : bindlessResourcesVec)
+            {
+                // Iterate through all the ResourceViews
+                for(const auto& resourceViewsIt : it.m_bindlessResources)
+                {
+                    //first = is the resource read only
+                    //second = native mtlresource pointer
+                    AZStd::pair<bool, id<MTLResource>> resourceInfo = GetResourceInfo(it.m_bindlessResourceType,
+                                                                     resourceViewsIt.get());
+                    
+                    if(resourceInfo.first)
+                    {
+                        m_untrackedResourcesComputeRead.insert(resourceInfo.second);
+                    }
+                    else
+                    {
+                        m_untrackedResourcesComputeReadWrite.insert(resourceInfo.second);
+                    }
+                }
             }
-            return true;
         }
-
+    
+        void CommandList::CollectBindlessGfxUntrackedResources(const ShaderResourceGroup* shaderResourceGroup,
+                                            ArgumentBuffer::ResourcesPerStageForGraphics& untrackedResourcesGfxRead,
+                                            ArgumentBuffer::ResourcesPerStageForGraphics& untrackedResourcesGfxReadWrite)
+        {
+            if(shaderResourceGroup->GetData().GetBindlessViewsSize()==0)
+            {
+                return;
+            }
+            
+            const AZStd::vector<RHI::ShaderResourceGroupData::BindlessResourceViews> bindlessResourcesVec =
+                                    shaderResourceGroup->GetData().GetAllBindlessViews();
+            for(const auto& it : bindlessResourcesVec)
+            {
+                // Iterate through all the ResourceViews
+                for(const auto& resourceViewsIt : it.m_bindlessResources)
+                {
+                    //first = is the resource read only
+                    //second = native mtlresource pointer
+                    AZStd::pair<bool, id<MTLResource>> resourceInfo = GetResourceInfo(it.m_bindlessResourceType,
```

Comment:
Nit: could this read more clearly with a well named type instead of `AZStd::pair<bool, id<MTLResource>>`